### PR TITLE
[Privatization] Handle Reference Method/Static Call from outside file on ChangeReadOnlyPropertyWithDefaultValueToConstantRector

### PR DESCRIPTION
**Pull request by samsonasik**
Closes #6214 by update `NodeRepository` to parse outside file if not found in the list of registered class method.

### CODE REVIEW

**Comment by samsonasik** [comment]
Also fix ReturnThisRemoveRector on check "vendor" dir

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: packages/NodeCollector/NodeCollector/NodeRepository.php [hunk 1]
```diff
@@ -223,6 +243,31 @@ public function findClassMethod(string $className, string $methodName): ?ClassMe
...
+        $fileContent = $this->smartFileSystem->readfile($fileName);
+        try {
+            $nodes = $this->parser->parse($fileContent);
+            return $this->getClassMethodFromNodes((array) $nodes, $classReflection, $className, $methodName);
+        } catch (Error $error) {
+            $isNotSyntaxError = ! Strings::contains($error->getMessage(), 'Syntax error');
+            if ($isNotSyntaxError) {
+                throw $error;
+            }
+        }
```
Comment (TomasVotruba) [inline packages/NodeCollector/NodeCollector/NodeRepository.php:269]: Why is this needed? It might hide unwated bugs in broken code

Comment (samsonasik) [reply packages/NodeCollector/NodeCollector/NodeRepository.php:269]: It got error on php < 8 on `FluentChainMethodCallToNormalMethodCallRector` tests: 

```bash
1) Rector\Tests\Defluent\Rector\MethodCall\FluentChainMethodCallToNormalMethodCallRector\FluentChainMethodCallToNormalMethodCallRectorTest::test with data set #17 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
59
PhpParser\Error: Syntax error, unexpected T_CONSTANT_ENCAPSED_STRING on line 208
```

ref https://github.com/rectorphp/rector/runs/2454146699#step:5:58

File: packages/NodeCollector/NodeCollector/NodeRepository.php [hunk 2]
```diff
@@ -499,6 +544,50 @@ public function findClassLike(string $classLikeName): ?ClassLike
...
+            }
+
+            $classMethod = $this->getClassMethod($node, $shortName, $methodName);
+            if ($classMethod instanceof ClassMethod) {
+                return $classMethod;
+            }
+        }
+
+        return null;
+    }
```
Comment (TomasVotruba) [inline packages/NodeCollector/NodeCollector/NodeRepository.php:577]: NodeRepository should only work with nodes in analysed project.

For parsing vendor/Reflection code, the `\Rector\Core\Reflection\FunctionLikeReflectionParser` should be used instead. Could you move the logic there and re-use content in that service?

Comment (samsonasik) [reply packages/NodeCollector/NodeCollector/NodeRepository.php:577]: I will try.

File: packages/NodeCollector/NodeCollector/NodeRepository.php (hunk 1 above)
Comment (TomasVotruba) [reply packages/NodeCollector/NodeCollector/NodeRepository.php:269]: What file is that?

Comment (samsonasik) [reply packages/NodeCollector/NodeCollector/NodeRepository.php:269]: in these 2 fixture files:

1. https://github.com/rectorphp/rector/blob/main/rules-tests/Defluent/Rector/MethodCall/FluentChainMethodCallToNormalMethodCallRector/Fixture/skip_date_time.php.inc

2. https://github.com/rectorphp/rector/blob/main/rules-tests/Defluent/Rector/MethodCall/FluentChainMethodCallToNormalMethodCallRector/Fixture/skip_date_time_modify_command.php.inc

Comment (samsonasik) [reply packages/NodeCollector/NodeCollector/NodeRepository.php:269]: Looking at the fixture files, it seems the erorr is caused by try to fetch to native class DateTime's method, so it may need `isBuiltIn()` check

Comment (samsonasik) [reply packages/NodeCollector/NodeCollector/NodeRepository.php]: Yes, it needs `isBuiltIn()` check https://github.com/rectorphp/rector/pull/6256/commits/7abde9909df1e8eebe23f8f2b61969d0f6ac8806 🎉

File: packages/NodeCollector/NodeCollector/NodeRepository.php (hunk 2 above)
Comment (samsonasik) [reply packages/NodeCollector/NodeCollector/NodeRepository.php]: implemented, moved to FunctionLikeReflectionParser https://github.com/rectorphp/rector/pull/6256/commits/677613048870fcdca38bd98fb5701cb791fea3ab
